region/crateria/west.json

```diff
@@ -799,24 +959,40 @@
           "to": [
             {
               "id": 2,
-              "requires": [
-                {"or": [
-                  "ScrewAttack",
-                  "canUseMorphBombs",
-                  {"and": [
+              "strats": [
+                {
+                  "name": "Base",
+                  "notable": false,
+                  "requires": ["ScrewAttack"]
+                },
+                {
+                  "name": "Morph Bomb Gauntlet Entrance (In)",
+                  "notable": true,
+                  "requires": ["canUseMorphBombs"]
+                },
+                {
+                  "name": "Power Bomb Gauntlet Entrance (In)",
+                  "notable": true,
+                  "requires": [
                     "Morph",
                     "PowerBomb",
                     {"ammo": {
                       "type": "PowerBomb",
                       "count": 3
                     }}
-                  ]},
-                  {"canComeInCharged": {
-                    "fromNode": 1,
-                    "framesRemaining": 0,
-                    "shinesparkFrames": 100
-                  }}
-                ]}
+                  ]
+                },
+                {
+                  "name": "Shinespark",
+                  "notable": false,
+                  "requires": [
+                    {"canComeInCharged": {
+                      "fromNode": 1,
+                      "framesRemaining": 0,
+                      "shinesparkFrames": 100
```

Comment:
There should be a number of runwaytiles field to this right?  You can make this spark in vanilla

Reply:
It may not be super instinctive, but there's no tile count for `canComeInCharged` because it would be redundant with things we've already entered.

See the documentation for `canComeInCharged` here, most notably the third bullet in additional considerations: https://github.com/miketrethewey/sm-json-data/blob/master/logicalRequirements.md.

In short, `canComeInCharged` can be fulfilled by combining the two runways attached to a door transition.

In this case, we have 6 tiles in the runway in this room and 10 in the adjacent runway. One of those is lost in the transition. This leaves us with a 15-tile runway with one open end, including 2 diagonal up tiles and one diagonal down tile. An algorithm would ideally gather all this data and decide whether that's doable under the current configuration.

I don't know how diagonals impact the charging mechanics, but I know they shorten the distance (and that up and down diagonals seem to shorten it by different amounts). That's why all the data is there in the runways. Maybe the topic has been studied by someone in the SM community? In the end, that's out of scope for this project, but it would be useful when writing the algorithm.

Reply:
Runways are part of the door definitions, got it.  I remembered them being a thing but couldn't remember where they'd be located.